airflow/providers/amazon/aws/operators/datasync.py

```diff
@@ -27,25 +25,45 @@
 from airflow.utils.decorators import apply_defaults
 
 
-class AWSDataSyncCreateTaskOperator(BaseOperator):
-    r"""Create an AWS DataSync Task.
+# pylint: disable=too-many-instance-attributes, too-many-arguments
+class AWSDataSyncOperator(BaseOperator):
+    r"""Find, Create, Update, Execute and Delete AWS DataSync Tasks.
+
+    If ``do_xcom_push`` is True, then the TaskArn and TaskExecutionArn which
+    were executed will be pushed to an XCom.
 
-    If there are existing Locations which match the specified
-    source and destination URIs then these will be used for the Task.
-    Otherwise, new Locations can be created automatically,
-    depending on input parameters.
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:AWSDataSyncOperator`
 
-    If ``do_xcom_push`` is True, the TaskArn which is created
-    will be pushed to an XCom.
+    .. note:: There may be 0, 1, or many existing DataSync Tasks. The default
+        behavior is to create a new Task if there are 0, or execute the Task
+        if there was 1 Task, or fail if there were many Tasks.
 
     :param str aws_conn_id: AWS connection to use.
-    :param str source_location_uri: Source location URI.
+    :param int wait_for_task_execution: Time to wait between two
+        consecutive calls to check TaskExecution status.
+    :param str task_arn: AWS DataSync TaskArn to use. If None, then this operator will
+        attempt to either search for an existing Task or create a new Task.
+    :param str source_location_uri: Source location URI to search for. All DataSync
+        Tasks with a LocationArn with this URI will be considered.
         Example: ``smb://server/subdir``
-    :param str destination_location_uri: Destination location URI.
+    :param str destination_location_uri: Destination location URI to search for.
+        All DataSync Tasks with a LocationArn with this URI will be considered.
         Example: ``s3://airflow_bucket/stuff``
-    :param bool case_sensitive_location_search: Whether or not to do a
+    :param bool location_search_case_sensitive: Whether or not to do a
         case-sensitive search for each Location URI.
-    :param dict create_task_kwargs: If no suitable TaskArn is found,
+    :param bool location_search_ignore_trailing_slash: Whether or not to strip
+        trailing / from locations when comparing them.
+    :param choose_task_callable: Optional callable which is passed a list
```

Comment:
Nice and extensible approach! However I think it deserves an example in examples section!

Reply:
Thanks, I thought it was neat since the underlying boto3 API will do validation anyway so may as well pass everything straight to it and we will be more future-proof. A useful technique for sure. 
There's an example already :)